app/models/piece.rb

```diff
@@ -176,8 +176,14 @@ def is_my_turn?
   end
   
   def valid_move?(x,y)
-    #i'm not sure I'm implementing in check correctly here. second set of eyes would be appreciated!
-    obstructed?(x,y) && off_the_board?(x,y) && in_check?(king) && not_moved_to_different_space?(x,y) && is_my_turn?
+    return false, 'move is obstructed' if obstructed?(x,y)
+    return false, 'move is off the board' unless off_the_board?(x,y) 
+    return false, 'king is in check' if in_check?(king) 
```

Comment:
One thing I notice right away is in_check? requires a king object but a king object isn't being passed in as one of the arguments. So imagine this will throw back a undefined method.

Reply:
I believe Ruby will return an array with the setup you have something like [false, 'move is obstructed'] or [false, 'move is obstructed']